e2e/johtoselvitys_ja_liite_hankkeelle.spec.ts

```diff
@@ -0,0 +1,256 @@
+  await expect(page.getByText('Hakemus tallennettu')).toBeVisible();
+  await expect(page.getByRole('heading', { name: 'Vaihe 5/5: Yhteenveto' })).toBeVisible();
+  await page.getByRole('button', { name: 'Lähetä hakemus' }).click();
+  await page.getByRole('button', { name: 'Vahvista' }).click();
+  await expect(page.getByText('Hakemus lähetetty')).toBeVisible();
+  await expect(page.getByTestId('allu_tunnus')).toBeVisible({ timeout: 30000 });
+  const hakemuksenTunnus = await page.getByTestId('allu_tunnus').textContent();
+  await expect(page.getByRole('button', { name: 'Peru hakemus' })).toBeVisible();
+  const linkkiHakemukseen = await page
+    .locator('a')
+    .filter({ hasText: /HAI/gm })
+    .getAttribute('href');
+  const linkkiHakemukseenEdit = linkkiHakemukseen?.slice(3);
+  const hakemusLinkki = `${testiData.testEnvUrl}${linkkiHakemukseenEdit}`;
+
+  // check allu
+  await page.goto(testiData.allu_url);
+  await expect(page.getByPlaceholder('Username')).toBeEmpty();
+  await page.getByPlaceholder('Username').click();
+  await page.getByPlaceholder('Username').fill(testiData.allupw);
+  await page.getByRole('button', { name: 'Submit' }).click();
+  await expect(page.getByRole('link', { name: 'HAKEMUKSET' })).toBeVisible();
+  await page.getByRole('link', { name: 'HAKEMUKSET' }).click();
+  await expect(page.getByRole('button', { name: 'HAE' })).toBeVisible();
+  await page.getByRole('button', { name: 'HAE' }).click();
+  await page.getByRole('link', { name: `${hakemuksenTunnus}` }).click();
+  await page.getByRole('button', { name: 'NÄYTÄ UUDET TIEDOT' }).click();
+  await page.getByRole('button', { name: 'KÄSITTELYYN' }).click();
+  await page.getByLabel('Hakemuksen lajit *').getByText('Hakemuksen lajit').click();
+  await page.getByText('Katu- ja vihertyöt').click();
+  await page.locator('.cdk-overlay-container > div:nth-child(3)').click();
+  await expect(page.getByRole('button', { name: 'TALLENNA' })).toBeVisible();
+  await page.getByRole('button', { name: 'TALLENNA' }).click();
+  await expect(page.getByRole('button', { name: 'PÄÄTTÄMISEEN' })).toBeVisible();
+  await page.getByRole('button', { name: 'PÄÄTTÄMISEEN' }).click();
+  await expect(page.getByRole('button', { name: 'PÄÄTÄ' })).toBeVisible();
+  await page.getByRole('button', { name: 'PÄÄTÄ' }).click();
+  await expect(page.getByRole('heading', { name: 'Päätä hakemus' })).toBeVisible();
+  await page.getByRole('button', { name: 'PÄÄTÄ' }).click();
+  await expect(page.getByRole('heading', { name: 'TYÖJONO' })).toBeVisible();
+
+  // Odotetaan tuloksia
+  await expect(async () => {
+    await page.goto(hakemusLinkki);
+    await page.getByText('Hakemukset').click();
+    await expect(page.getByTestId('application-status-tag')).toBeVisible();
+    await expect(page.getByTestId('application-status-tag')).toContainText('Päätös', {
+      timeout: 5000,
+    });
+  }).toPass({ intervals: [3000, 3000, 3000, 3000, 3000, 3000, 3000, 3000], timeout: 120000 });
```

Comment:
This could use the shared function.
```suggestion
   await tarkistaTulokset(page, hakemusLinkki, 'Päätös');
```

Reply:
No that's bad policy

Reply:
Sharing code when doing identical things is a best practice. Copy-pasting is bad practice.

Reply:
We discussed this and decided against reusing the status checking function. The `tarkistaTulokset` function should be moved to the file it's called from, since it's not meant to be shared between tests. Doesn't have to be in this PR.

Reply:
The function could have been moved to the `johtoselvitys_tilaus_taydennyspyynto.spec.ts` instead of unrolling the contents?